src/views/albums/Album.vue

```diff
@@ -102,6 +108,24 @@ export default {
     },
   },
   methods: {
+    ...mapActions("albums", ["read"]),
+    ...mapActions("tracks", { indexTracks: "index" }),
+    async fetchContent(newValue, oldValue) {
+      // After loading the content, the router will change the id from a string to a number
+      // but we don't actually want to load the content twice
+      if (newValue == oldValue) {
+        return;
+      }
+
+      const album = this.read(this.id);
+      const tracks = this.indexTracks(new TracksScope().album(this.id));
+      Promise.all([album, tracks]).finally(() => {
```

Comment:
Why not `return await` this, to make the control flow in the async method explicit?